docs/proposals/001-ai-gateway-proposal/proposal.md

```diff
@@ -0,0 +1,597 @@
+# Envoy AI Gateway
+
+## Proposal Status
+ ***Draft***
+
+## Table of Contents
+
+<!-- toc -->
+
+-   [Summary](#summary)
+-   [Goals](#goals)
+-   [Non-Goals](#non-goals)
+-   [Proposal](#proposal)
+    -   [Personas](#personas)
+    -   [Axioms](#axioms)
+    -   [AIGatewayRoute](#aigatewayroute)
+    -   [AIServiceBackend](#aiservicebackend)
+    -   [BackendSecurityPolicy](#backendsecuritypolicy)
+    -   [Token Usage based Rate Limiting](#token-usage-rate-limiting)
+    -   [Diagrams](#diagrams)
+
+<!-- /toc -->
```

Comment:
maybe "## Terminology" section here? e.g. the word `provider` is used everywhere in here but the relationship with the word "backend" might seem unclear.

Reply:
> Do we need a TOC? Docusaurus autogenerates one to the right hand side

the question is this is not for the website i guess @yuzisun

Reply:
ye this is not user facing doc, it is for contributor or whoever wants to understand the architecture design.